pulsar-broker/src/main/java/org/apache/pulsar/broker/namespace/NamespaceService.java

```diff
@@ -1445,23 +1451,44 @@ public PulsarClientImpl getNamespaceClient(ClusterDataImpl cluster) {
         });
     }
 
-    public Optional<NamespaceEphemeralData> getOwner(NamespaceBundle bundle) throws Exception {
-        // if there is no znode for the service unit, it is not owned by any broker
-        return getOwnerAsync(bundle).get(pulsar.getConfiguration().getMetadataStoreOperationTimeoutSeconds(), SECONDS);
-    }
-
     public CompletableFuture<Optional<NamespaceEphemeralData>> getOwnerAsync(NamespaceBundle bundle) {
+        if (ExtensibleLoadManagerImpl.isLoadManagerExtensionEnabled(config)) {
+            ExtensibleLoadManagerImpl extensibleLoadManager = ExtensibleLoadManagerImpl.get(loadManager.get());
+            return extensibleLoadManager.getOwnershipWithLookupDataAsync(bundle)
+                    .thenCompose(lookupData -> {
+                        if (lookupData.isPresent()) {
+                            return CompletableFuture.completedFuture(
+                                    Optional.of(lookupData.get().toNamespaceEphemeralData()));
+                        } else {
+                            return CompletableFuture.completedFuture(Optional.empty());
+                        }
+                    });
+        }
         return ownershipCache.getOwnerAsync(bundle);
     }
 
+    public boolean checkOwnershipPresent(NamespaceBundle bundle) throws Exception {
+        return checkOwnershipPresentAsync(bundle).get(pulsar.getConfiguration()
+                        .getMetadataStoreOperationTimeoutSeconds(), SECONDS);
+    }
+
+    public CompletableFuture<Boolean> checkOwnershipPresentAsync(NamespaceBundle bundle) {
+        if (ExtensibleLoadManagerImpl.isLoadManagerExtensionEnabled(config)) {
+            ExtensibleLoadManagerImpl extensibleLoadManager = ExtensibleLoadManagerImpl.get(loadManager.get());
+            return extensibleLoadManager.getOwnershipAsync(Optional.empty(), bundle)
```

Comment:
can we pass `Optional.empty() ` for system topics like bsc?

Reply:
Can you explain in detail?

Reply:
I don't think when using `checkOwnershipPresentAsync` method, we can get the topic info.

Reply:
I think we can remove `Optional<ServiceUnitId> topic` and pass just `bundleUnit` only to getOwnershipAsync, like the one below. Let the leader own all system namespace topics, and we don't auto-unload system topics.

```
private CompletableFuture<Optional<String>> getOwnershipAsync(ServiceUnitId bundleUnit) {
        final String bundle = bundleUnit.toString();
        CompletableFuture<Optional<String>> owner;
        if (isSystemNamespace(bundleUnit.toString())) {
            owner = serviceUnitStateChannel.getChannelOwnerAsync();
        } else {
            owner = serviceUnitStateChannel.getOwnerAsync(bundle);
        }
        return owner;
```

Reply:
Plz add a unit test case to cover this too.

Reply:
If all system namespace topics are owned by the leader, it might affect the transaction performance?

Reply:
Another option might be using a different namespace for the load manager topics. (e.g system-load-balance/) In this case, we need to schedule compaction separately.

Reply:
Again, if this is not for the system topics, I am good with the current code.

Reply:
Yes, adding another namespace will be better. Also, we shouldn't let users delete the system namespace, right?

Reply:
I dont think users need to delete system topics.

Reply:
Ok, then `checkOwnershipPresentAsync` will not be used for the system topics.

Reply:
Plz throw NotSupportedOperationException if the input bundle's namespace is the system namespace .

Reply:
Updated.